feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java

```diff
@@ -102,44 +99,43 @@ public void start(Stage stage) {
           // check for primary or secondary mouse click
           if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
 
+            // clear previous selections
+            featureLayer.clearSelection();
+
             // create a point from where the user clicked
             Point2D point = new Point2D(event.getX(), event.getY());
 
-            // convert to map coordinate
-            Point mapPoint = mapView.screenToLocation(point);
-
             // identify the clicked features
-            QueryParameters queryParameters = new QueryParameters();
-            queryParameters.setGeometry(GeometryEngine.buffer(mapPoint, 10));
-            final ListenableFuture<FeatureQueryResult> selectFeatureQuery = featureLayer.selectFeaturesAsync
-                (queryParameters, FeatureLayer.SelectionMode.NEW);
-            selectFeatureQuery.addDoneListener(() -> {
+            ListenableFuture<IdentifyLayerResult> identifyLayerResult =
+              mapView.identifyLayerAsync(featureLayer, point, 10, false, 1);
+            identifyLayerResult.addDoneListener(() -> {
 
               try {
-                FeatureQueryResult result = selectFeatureQuery.get();
+                IdentifyLayerResult identifiedLayer = identifyLayerResult.get();
+                List<GeoElement> identifiedLayerGeoElements = identifiedLayer.getElements();
                 // get the first selected feature
-                Iterator<Feature> iterator = result.iterator();
-                if (iterator.hasNext()) {
-                  ArcGISFeature selectedFeature = (ArcGISFeature) iterator.next();
+                for (var geoElement : identifiedLayerGeoElements) {
```

Comment:
For the sake of being extra clear which class we are dealing with here, we should write:
```suggestion
                for (GeoElement geoElement : identifiedLayerGeoElements) {
```

There's no harm in using `var` generally, but for samples we want to make sure our users know exactly which class they are dealing with at any point.

Reply:
I agree with Jon here: when using `var`, it's typically to improve readability in the case of a super long class name, or multiple blocks of repeated class types. In this for loop, it's better to err on the side of readability with `GeoElement`.

Reply:
Understood, thanks both for the explanation. I've changed that @Rachael-E